### PR TITLE
refactor: Move mib based polling into yaml config files

### PR DESCRIPTION
**Pull request by laf**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/)
- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

Moved mib polling calls into discover_device() function so they only need defining in the yaml definitions file now.

Updated docs to show new examples.

Updated html page to tell users they can enable per device + have per device check to show mib page.

Move code from `includes/discovery/os/*.inc.php` to `includes/definitions/*.yaml`.

### CODE REVIEW

**Comment by laf** [comment]
bump

File: doc/Extensions/MIB-based-polling.md [hunk 1]
```diff
@@ -147,32 +147,25 @@ graph.
...
 ## Configuration
 ### Main Configuration
-In `/opt/librenms/config.php` add `$config['poller_modules']['mib'] = 1;`
+In `/opt/librenms/config.php` add `$config['poller_modules']['mib'] = 1;` to enable MIB polling globally.
+Alternatively you can enable MIB polling per device by enabling it within the modules section for the specific device.
 
 ### Discovery
 
-You need to add your desired MIBs to `/opt/librenms/mibs` folder. Afterwards you need to register your MIBs to the discovery function. 
+You need to add your desired MIBs to `/opt/librenms/mibs` folder. Afterwards you need to register your MIBs to the OS definition.
```
Comment (murrant) [inline doc/Extensions/MIB-based-polling.md]: This better?
```
You need to add your desired MIBs to `/opt/librenms/mibs` folder. 
You will then need to create a listing of the OIDs you wish to use in the OS definition
```

**Comment by murrant** [comment]
Other than the small documentation change +1

**Comment by laf** [comment]
Doc updated.

**Pull request merged**